server/punishments.ts

```diff
@@ -679,6 +679,7 @@ export const Punishments = new class {
 		const ipStr = typeof user !== 'string' ? ` [${(user as User).latestIp}]` : '';
 		const roomid = typeof room !== 'string' ? (room as Room).roomid : room;
 		Rooms.global.modlog(`(${roomid}) AUTO${namelock ? `NAME` : ''}LOCK: [${userid}]${ipStr}: ${reason}`);
+		Chat.parse('/savereplay', Rooms.get(room), user as User, (user as User).connections[0]);
```

Comment:
This still isn't right, you can trigger this with a userid (ie if you nameblacklist someone with 2 RBs). You also need to make sure that `connections[0]` is actually defined (ie, punishing an offline user)

Reply:
I didn't even check whether the room was a battle at all. What was I thinking? I guess I'm dumb.